app/Components/App.js

```diff
@@ -1,17 +1,61 @@
 import React from 'react';
 
+import io from 'socket.io-client';
+let socket = io.connect('');
+
 class App extends React.Component {
   constructor(props) {
     super(props);
+    this.joinGame = this.joinGame.bind(this);
+    socket.on('assigngameID', (gameID) => this._assign_game_ID(gameID));
+    socket.on('isgameID', (flag) => this._is_game_ID(flag));
+    this.state = {value: "", warning_on: false};
+    this.handleChange = this.handleChange.bind(this);
+    this.joinGame = this.joinGame.bind(this);
+  }
+
+  joinGame(e) {
+    e.preventDefault();
+    socket.emit('joingame', this.state.value);
+  }
+
+  handleChange(event) {
+    this.setState({value: event.target.value, warning_on: false});
+  }
+
+  _assign_game_ID(gameID) {
```

Comment:
I'd recommend using keeping everything in the same camelCase style.  If this seems reasonable, you can add `"camelcase": ["error",  { "properties": "always" }],` to `.eslintrc` to help check this too ([docs](http://eslint.org/docs/rules/camelcase).  I added this to the teacher moments repo too: https://github.com/mit-teaching-systems-lab/threeflows/pull/214